kernel/driver.cc

```diff
@@ -244,8 +244,14 @@ int main(int argc, char **argv)
 	bool mode_q = false;
 
 #if defined(YOSYS_ENABLE_READLINE) || defined(YOSYS_ENABLE_EDITLINE)
-	if (getenv("HOME") != NULL) {
-		yosys_history_file = stringf("%s/.yosys_history", getenv("HOME"));
+	if (getenv("XDG_STATE_HOME") == NULL || getenv("XDG_STATE_HOME")[0] == '\0') {
+		if (getenv("HOME") != NULL) {
+			yosys_history_file = stringf("%s/.local/state/.yosys_history", getenv("HOME"));
```

Comment:
I think this should probably not be a dotfile when it's under `~/.local/state`:

```suggestion
			yosys_history_file = stringf("%s/.local/state/yosys_history", getenv("HOME"));
```

Reply:
Also, what happens if there is no `~/.local/state` directory? It should probably be created if it doesn't exist.

Reply:
It seems that `~/.local/state` is often nonexistent. If we are creating a directory here already, it's best to create `~/.local/state/yosys/` and then put a file named `history` there.